File: src/main/java/bbarena/model/dice/JavaRandomizer.java
package bbarena.model.dice;

import java.security.SecureRandom;
import java.util.HashMap;
import java.util.Map;
import java.util.Random;
import java.util.UUID;

/**
 * 
 * @author cammerucci
 *
 */
public class JavaRandomizer implements DieRandomizer{

	private final Random _random = new SecureRandom();
	private final Map<String, Integer> _results;
	
	public JavaRandomizer() {
		_results = new HashMap<>();
	}

	public int getRollFace(int facesNumber) {
		return getRollFace(facesNumber, null);
	}

	public int getRollFace(int facesNumber, String id) {
		Integer res = null;

		if (id != null) {
			res = _results.get(id);
		}
		if (res == null) {
			res = _random.nextInt(facesNumber - 1) + 1;
			if (id != null) {
				_results.put(id, res);
			}
		}
		
		return res;
	}
}

File: src/main/kotlin/bbarena/model/dice/JavaRandomizer.kt
package bbarena.model.dice

import java.security.SecureRandom

/**
 *
 * @author cammerucci
 */
class JavaRandomizer : DieRandomizer {

    private val random = SecureRandom()
    private val results = mutableMapOf<String, Int>()

    fun getRollFace(facesNumber: Int): Int {
        return getRollFace(facesNumber, "")
    }

    override fun getRollFace(faces: Int, rollId: String): Int {
        return results[rollId] ?: {
            val res = random.nextInt(faces - 1) + 1
            results[rollId] = res
            res
        }.invoke()
    }
}
